rfcs/0001-messages.md

```diff
@@ -0,0 +1,412 @@
+
+### Header
+
+The `Header` part of the message is 4 bytes total:
+
+Bytes | Field name | Data type
+------|------------|-----------
+1     | msg_type   | u8
+1     | version    | u8
+1     | sender     | u8
+1     | receiver   | u8
+
+### Primitive types
+
+`Payload`s use data types that we need to specify first. We have 3 primitive types inside the payload messages:
+
+#### `Scalar`
+
+`Scalar` is a an alias for `jubjub::Fr`. We use `Scalar::to_bytes` and `Scalar::from_bytes` to get a 32-byte little-endian canonical representation. See https://github.com/zkcrypto/bls12_381/blob/main/src/scalar.rs#L252
+
+#### `AffinePoint`
+
+Much of the math in FROST is done using `jubjub::ExtendedPoint`. But for message exchange `jubjub::AffinePoint`s are a better choice, as their byte representation is smaller.
+
+Conversion from one type to the other is trivial:
+
+https://docs.rs/jubjub/0.6.0/jubjub/struct.AffinePoint.html#impl-From%3CExtendedPoint%3E
+https://docs.rs/jubjub/0.6.0/jubjub/struct.ExtendedPoint.html#impl-From%3CAffinePoint%3E
+
+We use `AffinePoint::to_bytes` and `AffinePoint::from_bytes` to get a 32-byte little-endian canonical representation. See https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L443
+
+#### VerificationKey
+
+`VerificationKey<SpendAuth>`s can be serialized and deserialized using `<[u8; 32]>::from` and `VerificationKey::from`. See https://github.com/ZcashFoundation/redjubjub/blob/main/src/verification_key.rs#L80-L90 and https://github.com/ZcashFoundation/redjubjub/blob/main/src/verification_key.rs#L114-L121.
+
+### Payload
+
+Payload part of the message is variable in size and depends on message type.
+
+#### `SharePackage`
+
+Bytes           | Field name       | Data type
+----------------|------------------|-----------
+32              | secret_share     | Scalar
+1               | participants     | u8
+32*participants | share_commitment | Vec\<AffinePoint\>
+32              | group_public     | VerificationKey<SpendAuth>
+
+#### `SigningCommitments`
+
+Bytes   | Field name          | Data type
+--------|---------------------|-----------
+32   | hiding | AffinePoint
+32   | binding | AffinePoint
+
+#### `SigningPackage`
+
+Bytes                  | Field name         | Data type
+-----------------------|--------------------|-----------
+8                      | message_length     | u64
+message_length         | message            | Vec\<u8\>
+1                      | participants       | u8
+(1+32+32)*participants | signing_commitments| HashMap<ParticipantId, SigningCommitments>
+
+
+#### `SignatureShare`
+
+Bytes | Field name | Data type
+------|------------|-----------
+32    | signature  | Scalar
+
+#### `AggregateSignature`
+
+Bytes | Field name       | Data type
+------|------------------|-----------
+32    | group_commitment | AffinePoint
+32    | schnorr_signature| Scalar
+
+## Not included
```

Comment:
TODO: Add a section on persistent key material
- which key material is safe to re-use?
- how do users serialize and de-serialize secret keys?

Reply:
@oxarbitrage please make a ticket for this TODO before you merge.

Reply:
https://github.com/ZcashFoundation/redjubjub/issues/113